src/api/audio.c

```diff
@@ -68,8 +68,14 @@ int l_lovrAudioNewSource(lua_State* L) {
     stream = *type;
   } else {
     Blob* blob = luax_readblob(L, 1, "Source");
+    if (!blob->data) // readblob (1) never returns NULL (2) is responsible for setting Lua error
+      return 0;
```

Comment:
I'm still confused: given `readblob` either returns a Blob or throws an error, we shouldn't need to check the return value and could just omit this `if`.

Reply:
You are right, I messed up in my initial testing and thought this was necessary.